pywps/app/Service.py

```diff
@@ -301,12 +301,10 @@ def execute(self, identifier, wps_request, uuid):
         except KeyError:
             raise InvalidParameterValue("Unknown process '%r'" % identifier, 'Identifier')
 
-        olddir = os.path.abspath(os.curdir)
         try:
-            os.chdir(process.workdir)
             response = self._parse_and_execute(process, wps_request, uuid)
         finally:
-            os.chdir(olddir)
+            process.clean()
```

Comment:
If the case of an async job isn't this going to remove the `workdir` before the job has actually run?

The [ExecuteResponse object](https://github.com/geopython/pywps/blob/bdd404e7af9cac7e343e3ebf32111a3d43dd8137/pywps/response/execute.py) is currently responsible for calling `clean` on the process. So what's the reason for calling `clean` here?

Reply:
oh my! my "reason" for cleaning up here, is that its in the same area of code as the creation - but by adding async - and having an ExecuteResponse cleanup, its redundant.